packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -2,21 +2,63 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:pool/pool.dart';
+
 import '../../artifacts.dart';
+import '../../asset.dart';
 import '../../base/file_system.dart';
 import '../../base/io.dart';
+import '../../base/process.dart';
 import '../../base/process_manager.dart';
 import '../../build_info.dart';
+import '../../devfs.dart';
 import '../../globals.dart';
-import '../../macos/cocoapods.dart';
-import '../../project.dart';
+import '../../macos/xcode.dart';
 import '../build_system.dart';
-import '../exceptions.dart';
-import 'assets.dart';
 import 'dart.dart';
 
 const String _kOutputPrefix = '{PROJECT_DIR}/macos/Flutter/ephemeral/FlutterMacOS.framework';
 
+/// The copying logic for flutter assets in macOS.
+// TODO(jonahwilliams): remove once build planning lands.
+class MacOSAssetBehavior extends SourceBehavior {
+  const MacOSAssetBehavior();
+
+  @override
+  List<File> inputs(Environment environment) {
+    final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
+    assetBundle.build(
+      manifestPath: environment.projectDir.childFile('pubspec.yaml').path,
+      packagesPath: environment.projectDir.childFile('.packages').path,
+    );
+    // Filter the file type to remove the files that are generated by this
+    // command as inputs.
+    final List<File> results = <File>[];
+    final Iterable<DevFSFileContent> files = assetBundle.entries.values.whereType<DevFSFileContent>();
+    for (DevFSFileContent devFsContent in files) {
+      results.add(fs.file(devFsContent.file.path));
+    }
+    return results;
+  }
+
+  @override
+  List<File> outputs(Environment environment) {
+    final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
+    assetBundle.build(
+      manifestPath: environment.projectDir.childFile('pubspec.yaml').path,
+      packagesPath: environment.projectDir.childFile('.packages').path,
+    );
+    final String prefix = fs.path.join(environment.projectDir.path, 'macos',
```

Comment:
Why is the macos/Flutter/ephemeral part of this hard-coded rather than pulled from the project getter?

Reply:
Updated all to use FlutterProject